recipes/full_finetune_distributed.py

```diff
@@ -697,15 +698,24 @@ def train(self) -> None:
                 # Compute loss
                 # Loss is normalized by default so we multiply by the number of tokens
                 # This way we can normalize by the total number of tokens if we're accumulating gradients
-                running_loss += self._loss_fn(logits, labels) * current_num_tokens
+                current_loss = self._loss_fn(logits, labels) * current_num_tokens
 
                 # free logits otherwise it peaks backward memory
                 del logits
 
+                running_loss += current_loss
```

Comment:
i dont know if this doubles memory, because it creates a copy of the the current_loss. If it does (you can test with a large enough batch_size), then maybe we should just do it if self._gradient_accumulation_steps>1

Reply:
Can you clarify? The loss itself is a single float, right? Do you mean that we are now keeping additional activations around that we otherwise wouldn't?

Reply:
The loss is a single float, so if the only extra memory is that, then ignore my comment. But I dont know if maybe because of hooks or something else, it ends up not releasing memory or consuming more than just that float.

Reply:
Yeah agreed, will eventually kick off full runs to check peak memory